codec/src/codec.rs

```diff
@@ -63,363 +61,23 @@ pub trait SizedCodec: Codec {
             N
         );
```

Comment:
If we moved `LEN_ENCODED` to be a generic parameter, would that impact some meaningful part of the UX?

Reply:
If I understand you correctly, this is possible, but the same issue as here comes up
https://github.com/commonwarexyz/monorepo/pull/603#discussion_r1996369132

Reply:
Ah got it. Figured this was one of those strings not to pull...